spec/datadog/appsec/contrib/active_record/patcher_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/contrib/active_record/patcher'
+
+RSpec.describe Datadog::AppSec::Contrib::ActiveRecord::Patcher do
+  describe '#prepended_class_name' do
+    context 'when ActiveRecord version is 7.1 or higher' do
+      before do
+        stub_const(
+          '::ActiveRecord',
+          Module.new do
+            module_function def gem_version
+              Gem::Version.new('7.1')
+            end
+          end
+        )
+      end
+
+      it 'returns Instrumentation::InternalExecQueryAdapterPatch' do
+        expect(described_class.prepended_class_name(:postgresql)).to eq(
+          Datadog::AppSec::Contrib::ActiveRecord::Instrumentation::InternalExecQueryAdapterPatch
+        )
+      end
+    end
+
+    context 'when ActiveRecord version is lower than 7.1' do
+      before do
+        stub_const(
+          '::ActiveRecord',
+          Module.new do
+            module_function def gem_version
+              Gem::Version.new('7.0')
+            end
+          end
+        )
```

Comment:
Maybe a matter of preference, but I find it easier to read

```suggestion
        stub_const('::ActiveRecord', Module.new do
          module_function def gem_version
            Gem::Version.new('7.0')
          end
        end)
```